cms/envs/common.py

```diff
@@ -387,6 +387,23 @@
     # .. toggle_status: supported
     # .. toggle_warnings: None
     'ENABLE_ORA_USER_STATE_UPLOAD_DATA': False,
+
+    # .. toggle_name: DEPRECATE_OLD_COURSE_KEYS_IN_STUDIO
+    # .. toggle_implementation: DjangoSetting
+    # .. toggle_default: True
+    # .. toggle_description: Warn about removing support for deprecated course keys.
+    #      To enable, set to True.
+    #      To disable, set to False.
+    #      To enable with a custom support deadline, set to an ISO-8601 date string:
+    #        eg: '2020-09-01'
```

Comment:
Aww, that's one of my favorite parts of dynamic languages :)

I generally try to minimize the number of configuration values needed (eg: so you don't toggle a feature back on but realize it depends on some other now-outdated config value [1]).

But I'm aware this position isn't non-controversial :)

## Anecdotes
- [1] Stanford once enabled some feature flag (PDF receipts for old `shoppingcart` maybe?), only to later find that the feature was happy to run with its, _aherm_, "unprofessional" default values (part of it was literally the "ALL YOUR BASE ARE BELONG TO US!" text). That made for a fun explanation :)

Reply:
Yeah, the place where it's bitten us is where people forget about the other possible types and it just starts blowing up on certain edge cases because you're calling a datetime method on something that could also be a string, etc. But like I said, not going to try to argue it for this one. :)